ui/store/actions.js

```diff
@@ -3172,8 +3172,10 @@ export function getContractMethodData(data = '') {
   return (dispatch, getState) => {
     const prefixedData = addHexPrefix(data);
     const fourBytePrefix = prefixedData.slice(0, 10);
+    if (fourBytePrefix.length < 10) {
+      return Promise.resolve({});
+    }
```

Comment:
Nice fix here. 

I'm a bit confused as to why we did support `fourBytePrefix === '0x'` on line 3183 though. I think we'll either want to continue supporting this or remove the line.

Reply:
That's a great call out @digiwand, I'm not sure what that is about. @FrederikBolding any idea why we would expect `0x` to be present in `knownMethodData` and why we specify it here in the if block on lines 3180-3183?

Reply:
@danjm @danfinlay do either of you happen to know why we carved out an exception for `0x` here?

Reply:
Looks like this logic was added here https://github.com/MetaMask/metamask-extension/pull/9048 cc: @Gudahtt 

If I'm understanding this right, we were merely preventing calls to check the 4byte directory if `fourBytePrefix === 0x`. This should mean we can remove the `fourBytePrefix === 0x` check with this new logic you added in this PR

Reply:
done [here](https://github.com/MetaMask/metamask-extension/pull/15473/commits/53018843fb262a279034442d5430b0bbf39bac39) @digiwand 😄

Reply:
Yes, @digiwand is correct, as far as I can recall anyway.